### PR TITLE
Add Test Coverage for Transformers import

### PR DESCRIPTION
**Pull request by abarbosa94**
### Description
- This PR adds a simple set of unit tests to properly cover transformers' import use cases

### CODE REVIEW

**Comment by aaarrti** [comment]
Do I understand correctly, that you're using `pytest-mock` to simulate the case when `transformers` package is installed / is not installed? I'd recommend to create different `tox` environments for this, instead of mocking parts of pythons import logic.

File: quantus/helpers/model/pytorch_model.py [hunk 1]
```diff
@@ -108,25 +108,27 @@ def _get_model_with_linear_top(self) -> torch.nn:
...
+                        isinstance(x, dict)
+                        and ("input_ids" in x and "attention_mask" in x)
+                    )
+                ):
+                    raise ValueError(
+                        "When using HuggingFace pretrained models, please use Tokenizers output for `x` "
+                        "or make sure you're passing a dict with input_ids and attention_mask as keys"
+                    )
+                pred = self.model(**x, **model_predict_kwargs).logits
+                if self.softmax:
```
Comment (aaarrti) [inline quantus/helpers/model/pytorch_model.py:128]: Are you 100% sure that HF models never have softmax?

File: tests/functions/test_pytorch_model.py [hunk 2]
```diff
@@ -286,8 +293,40 @@ def test_add_mean_shift_to_first_layer(load_mnist_model):
...
+)
+def test_predict_transformers_installed(
+    mocker, transformers_installed, base_class, expected
+):
+    mocker.patch("importlib.util.find_spec", return_value=transformers_installed)
+    from quantus.helpers.model import pytorch_model
+
+    reload(pytorch_model)
+    # Mock the model's behavior
+    model_instance = PyTorchModel(model=mocker.MagicMock(spec=base_class))
```
Comment (artem-sereda) [inline tests/functions/test_pytorch_model.py:323]: Hi @abarbosa94, 

could you please make sure the original state of module/classes is restored after this function exits? You could do this with a context manager, e.g., https://github.com/understandable-machine-intelligence-lab/Quantus/blob/main/tests/metrics/conftest.py

Comment (aaarrti) [inline tests/functions/test_pytorch_model.py:323]: Hi @abarbosa94, 

could you please make sure the original state is restored after this function exits? This can be done using context-manager fixture, e.g., https://github.com/understandable-machine-intelligence-lab/Quantus/blob/main/tests/metrics/conftest.py

**Comment by aaarrti** [comment]
@abarbosa94, I believe that the test failures are caused by conflicting mocks. In case not, please don't spend too much time debugging those, i.e. this is a green light from me to just delete the failing tests (and unnecessary overcomplicated mocks)

**Comment by annahedstroem** [comment]
> @abarbosa94, I believe that the test failures are caused by conflicting mocks. In case not, please don't spend too much time debugging those, i.e. this is a green light from me to just delete the failing tests (and unnecessary overcomplicated mocks)

please see my answer on Slack! perhaps easiest and safest if you delete those mocks and dependencies? perhaps just push to this PR

File: quantus/helpers/model/pytorch_model.py (hunk 1 above)
Comment (abarbosa94) [reply quantus/helpers/model/pytorch_model.py:128]: I'd say yes: https://discuss.huggingface.co/t/softmax-vs-logits/7008

Based on this I'd say it returns logits by design
